### PR TITLE
Sprint 6 development > master PR

### PR DESCRIPTION
**Pull request by Michelbaartman**
https://trello.com/b/WIO3vVmc/robot-claw
Vertical slice 5
https://trello.com/c/UhcSvabb/50-vs5-robot-claw-slice

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,36 +1,30 @@
...
-    Claw claw(conn, touchSensorLeft, touchSensorRight);
+    UARTLib::HardwareUART conn(115200, UARTLib::UARTController::ONE, false);
+    Claw claw(conn, gripSensor);
 
     conn.begin();
-
     hwlib::wait_ms(1000);
 
-    // Check if the uArm is connected.
+    /// Check if the uArm is connected.
```
Comment (wvanbreukelen) [inline src/main.cpp]: Use ///< style comments

File: src/claw.cpp [hunk 2]
```diff
@@ -123,4 +151,26 @@ ClawState Claw::getState() {
     default:
         return ClawState::UNKNOWN;
     }
-}
\ No newline at end of file
+}
+
+ClawFeedback Claw::decodeGcodeResponse(char *response, size_t responseSize) {
```
Comment (wvanbreukelen) [inline src/claw.cpp:156]: Not using the ```responseSize``` parameter.

File: src/claw.cpp [hunk 3]
```diff
@@ -95,24 +115,32 @@ int Claw::receiveGcodeResponse(char *response, size_t responseSize, unsigned int
...
+
+        char response[15];
+        receiveGcodeResponse(response, 15);
+
+        /// Response format: $n ok V1\n
+        /// Search for V character
+        int vStart = 0;
+
+        for (int i = 0; i < 15; i++) {
+            /// If we have the Gcode response `$n ok V1` we only want to return the stuff behind the V mark (in this
```
Comment (wvanbreukelen) [inline src/claw.cpp]: Use ///< style comments

File: src/claw.cpp [hunk 4]
```diff
@@ -95,24 +115,32 @@ int Claw::receiveGcodeResponse(char *response, size_t responseSize, unsigned int
...
-            break;
+    if (!clawSensing.getState()) {
+        return ClawState::CLOSED;
+    } else {
+        uartComm << "#n P2232\n";
+
+        char response[15];
+        receiveGcodeResponse(response, 15);
+
+        /// Response format: $n ok V1\n
```
Comment (wvanbreukelen) [inline src/claw.cpp]: Use ///< style comments

File: src/claw.cpp [hunk 5]
```diff
@@ -73,8 +92,9 @@ int Claw::receiveGcodeResponse(char *response, size_t responseSize, unsigned int
                 responseCharCounter += 1;
 
                 lastRead = hwlib::now_us();
-            } else if (responseCharCounter > 0) { /// We have found a endline. If the response char counter is larger then zero
-                                                  /// (there is data), we will stopping polling for new data.
+            } else if (responseCharCounter > 0) {
+                /// We have found a endline. If the response char counter is larger then zero
```
Comment (wvanbreukelen) [inline src/claw.cpp]: Use ///< style comments

**Review by wvanbreukelen — CHANGES_REQUESTED** [review]
Some small changes to be made.

**Comment by ghost** [comment]
@wvanbreukelen We processed your feedback.

**Review by wvanbreukelen — APPROVED** [review]

**Pull request merged**